Scrolling by pressing space bar has regressed in trunk, is choppy (not 120Hz)

rr-codes opened the pull request:
#### cb5925ef6a3e94f82e753b50298c2a25f7160da0
<pre>
Scrolling by pressing space bar has regressed in trunk, is choppy (not 120Hz)
<a href="https://bugs.webkit.org/show_bug.cgi?id=246878">https://bugs.webkit.org/show_bug.cgi?id=246878</a>
rdar://101052130

Reviewed by Simon Fraser.

Moves keyboard smooth scrolling onto the scrolling thread instead of the main thread.

A new `ScrollAnimation` is created to facilitate this, `ScrollAnimationKeyboard`. Most
of the changes are to allow the state to propogate through the various classes and
the scrolling tree.

Since keyboard scrolling is now on a different thread, some tests had to be modified to
enable the `AsyncOverflowScrollingEnabled` and `AsyncFrameScrollingEnabled` settings.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::requestStartKeyboardAnimation):
(WebCore::FrameView::requestFinishKeyboardAnimation):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::requestStartKeyboardAnimation):
(WebCore::AsyncScrollingCoordinator::requestFinishKeyboardAnimation):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
(WebCore::ScrollingCoordinator::requestStartKeyboardAnimation):
(WebCore::ScrollingCoordinator::requestFinishKeyboardAnimation):
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
* Source/WebCore/page/scrolling/ScrollingStateScrollingNode.cpp:
(WebCore::ScrollingStateScrollingNode::ScrollingStateScrollingNode):
(WebCore::ScrollingStateScrollingNode::setKeyboardScrollData):
(WebCore::ScrollingStateScrollingNode::setKeyboardScrollEndData):
* Source/WebCore/page/scrolling/ScrollingStateScrollingNode.h:
(WebCore::ScrollingStateScrollingNode::keyboardScrollData const):
(WebCore::ScrollingStateScrollingNode::keyboardScrollEndData const):
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp:
(WebCore::ScrollingTreeScrollingNode::commitStateAfterChildren):
(WebCore::ScrollingTreeScrollingNode::handleKeyboardScrollRequest):
(WebCore::ScrollingTreeScrollingNode::handleKeyboardScrollEndRequest):
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.h:
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.h:
(WebCore::ScrollingTreeScrollingNodeDelegate::handleKeyboardScrollRequest):
(WebCore::ScrollingTreeScrollingNodeDelegate::handleKeyboardScrollEndRequest):
* Source/WebCore/page/scrolling/mac/ScrollingTreeScrollingNodeDelegateMac.h:
* Source/WebCore/page/scrolling/mac/ScrollingTreeScrollingNodeDelegateMac.mm:
(WebCore::ScrollingTreeScrollingNodeDelegateMac::handleKeyboardScrollRequest):
(WebCore::ScrollingTreeScrollingNodeDelegateMac::handleKeyboardScrollEndRequest):
* Source/WebCore/platform/KeyboardScrollingAnimator.cpp:
(WebCore::KeyboardScrollingAnimator::KeyboardScrollingAnimator):
(WebCore::KeyboardScrollingAnimator::beginKeyboardScrollGesture):
(WebCore::KeyboardScrollingAnimator::handleKeyUpEvent):
(WebCore::KeyboardScrollingAnimator::stopScrollingImmediately):
(WebCore::perpendicularAbsoluteUnitVector): Deleted.
(WebCore::KeyboardScrollingAnimator::updateKeyboardScrollPosition): Deleted.
(WebCore::farthestPointInDirection): Deleted.
(WebCore::KeyboardScrollingAnimator::stopKeyboardScrollAnimation): Deleted.
* Source/WebCore/platform/KeyboardScrollingAnimator.h:
* Source/WebCore/platform/ScrollAnimation.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/ScrollAnimation.h:
* Source/WebCore/platform/ScrollAnimationKeyboard.cpp: Added.
(WebCore::perpendicularAbsoluteUnitVector):
(WebCore::boxSideForDirection):
(WebCore::farthestPointInDirection):
(WebCore::ScrollAnimationKeyboard::scrollableDirectionsFromPosition):
(WebCore::ScrollAnimationKeyboard::ScrollAnimationKeyboard):
(WebCore::ScrollAnimationKeyboard::retargetActiveAnimation):
(WebCore::ScrollAnimationKeyboard::serviceAnimation):
(WebCore::ScrollAnimationKeyboard::startKeyboardScroll):
(WebCore::ScrollAnimationKeyboard::stopKeyboardScrollAnimation):
(WebCore::ScrollAnimationKeyboard::finishKeyboardScroll):
(WebCore::ScrollAnimationKeyboard::animateScroll):
(WebCore::ScrollAnimationKeyboard::debugDescription const):
* Source/WebCore/platform/ScrollAnimationKeyboard.h: Copied from Source/WebCore/platform/ScrollAnimation.cpp.
* Source/WebCore/platform/ScrollAnimator.cpp:
(WebCore::ScrollAnimator::ScrollAnimator):
* Source/WebCore/platform/ScrollableArea.cpp:
(WebCore::ScrollableArea::beginKeyboardScroll):
(WebCore::ScrollableArea::endKeyboardScroll):
* Source/WebCore/platform/ScrollableArea.h:
(WebCore::ScrollableArea::requestStartKeyboardAnimation):
(WebCore::ScrollableArea::requestFinishKeyboardAnimation):
* Source/WebCore/platform/ScrollingEffectsController.cpp:
(WebCore::ScrollingEffectsController::animationCallback):
(WebCore::ScrollingEffectsController::startKeyboardScroll):
(WebCore::ScrollingEffectsController::finishKeyboardScroll):
(WebCore::ScrollingEffectsController::updateKeyboardScrollingAnimatingState):
(WebCore::ScrollingEffectsController::scrollAnimationWillStart):
* Source/WebCore/platform/ScrollingEffectsController.h:
(WebCore::ScrollingEffectsControllerClient::updateKeyboardScrollPosition): Deleted.
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::requestStartKeyboardAnimation):
(WebCore::RenderLayerScrollableArea::requestFinishKeyboardAnimation):
* Source/WebCore/rendering/RenderLayerScrollableArea.h:

Canonical link: <a href="https://commits.webkit.org/256266@main">https://commits.webkit.org/256266@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fc8ff7d1afa2f9d08a3c794b419907fd149b9983

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95216 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4461 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28137 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104813 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/165072 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99207 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4469 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33220 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87534 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100708 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100883 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/3263 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/81766 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/30229 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/85166 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/86996 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73131 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38945 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18554 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36761 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19837 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4328 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40699 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/42535 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42684 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39079 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/platform/ScrollAnimationKeyboard.cpp:
```diff
@@ -0,0 +1,250 @@
+ScrollAnimationKeyboard::~ScrollAnimationKeyboard() = default;
+
+bool ScrollAnimationKeyboard::retargetActiveAnimation(const FloatPoint&)
+{
```
I still need to figure out what should go here

rr-codes commented on Source/WebCore/platform/ScrollableArea.cpp:
```diff
@@ -132,6 +132,22 @@ bool ScrollableArea::scroll(ScrollDirection direction, ScrollGranularity granula
 
+void ScrollableArea::beginKeyboardScroll(const KeyboardScroll& scrollData)
+{
+    ALWAYS_LOG_WITH_STREAM(stream << "RR_CODES " << __PRETTY_FUNCTION__);
```
remove log

rr-codes commented on Source/WebCore/platform/ScrollableArea.cpp:
```diff
@@ -132,6 +132,22 @@ bool ScrollableArea::scroll(ScrollDirection direction, ScrollGranularity granula
+
+void ScrollableArea::endKeyboardScroll(bool immediate)
+{
+    ALWAYS_LOG_WITH_STREAM(stream << "RR_CODES " << __PRETTY_FUNCTION__);
```
remove log

rr-codes commented on Source/WebCore/platform/ScrollingEffectsController.cpp:
```diff
@@ -62,7 +63,7 @@ void ScrollingEffectsController::animationCallback(MonotonicTime currentTime)
 
     updateRubberBandAnimatingState();
-    updateKeyboardScrollingAnimatingState(currentTime);
+//    updateKeyboardScrollingAnimatingState(currentTime);
```
remove

rr-codes commented on Source/WebCore/platform/ScrollingEffectsController.cpp:
```diff
@@ -483,12 +501,10 @@ void ScrollingEffectsController::stopScrollSnapAnimation()
 }
 
-void ScrollingEffectsController::updateKeyboardScrollingAnimatingState(MonotonicTime currentTime)
+void ScrollingEffectsController::updateKeyboardScrollingAnimatingState(MonotonicTime)
```
this isn't needed anymore

rr-codes commented on Source/WebCore/platform/ScrollAnimationKeyboard.cpp:
```diff
@@ -0,0 +1,250 @@
+
+String ScrollAnimationKeyboard::debugDescription() const
+{
+    return ""_s;
```
complete this implementation

smfr commented on Source/WebCore/page/FrameView.h:
```diff
@@ -269,6 +269,10 @@ class FrameView final : public ScrollView {
     void updateCompositingLayersAfterScrolling() final;
     static WEBCORE_EXPORT bool scrollRectToVisible(const LayoutRect& absoluteRect, const RenderObject&, bool insideFixed, const ScrollRectToVisibleOptions&);
 
+    bool requestStartKeyboardAnimation(const KeyboardScroll&) final;
```
Let's call this `requestStartKeyboardScrollAnimation`

smfr commented on Source/WebCore/page/FrameView.h:
```diff
@@ -269,6 +269,10 @@ class FrameView final : public ScrollView {
 
+    bool requestStartKeyboardAnimation(const KeyboardScroll&) final;
+
+    bool requestFinishKeyboardAnimation(bool immediate) final;
```
Remove the blank line above, and rename to `requestFinishKeyboardScrollAnimation`. Is it really "finish", or is it more like "stop"? I think stop would be better.

smfr commented on Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
```diff
@@ -254,6 +254,42 @@ void AsyncScrollingCoordinator::frameViewRootLayerDidChange(FrameView& frameView
+
+    stateNode->setKeyboardScrollData(scrollData);
+    // FIXME: This should schedule a rendering update
+    commitTreeStateIfNeeded();
```
This should not be necessary. I'm not sure why we have it for the other ones.

smfr commented on Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
```diff
@@ -254,6 +254,42 @@ void AsyncScrollingCoordinator::frameViewRootLayerDidChange(FrameView& frameView
+    if (!stateNode)
+        return false;
+
+    stateNode->setKeyboardScrollEndData(immediate);
```
Might be nicer to just encode the "stop" command in `KeyboardScroll` and use a single setter (which is what the "request animated scroll" stuff does).

smfr commented on Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
```diff
@@ -254,6 +254,42 @@ void AsyncScrollingCoordinator::frameViewRootLayerDidChange(FrameView& frameView
+
+    stateNode->setKeyboardScrollEndData(immediate);
+    // FIXME: This should schedule a rendering update
+    commitTreeStateIfNeeded();
```
This should not be necessary.

smfr commented on Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
```diff
@@ -102,6 +102,10 @@ class AsyncScrollingCoordinator : public ScrollingCoordinator {
 
+    WEBCORE_EXPORT bool requestStartKeyboardAnimation(ScrollableArea&, const KeyboardScroll&) final;
+
+    WEBCORE_EXPORT bool requestFinishKeyboardAnimation(ScrollableArea&, bool) final;
```
No blank line between them (since they are closely related).

smfr commented on Source/WebCore/page/scrolling/ScrollingCoordinator.h:
```diff
@@ -26,6 +26,7 @@
 #pragma once
 
 #include "EventTrackingRegions.h"
+#include "KeyboardScroll.h"
```
You should just forward declare, to avoid `#includes`.

smfr commented on Source/WebCore/page/scrolling/ScrollingStateScrollingNode.h:
```diff
@@ -27,6 +27,7 @@
 
 #if ENABLE(ASYNC_SCROLLING)
 
+#include "KeyboardScroll.h"
```
Prefer forward declaration.

smfr commented on Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp:
```diff
@@ -106,6 +106,12 @@ void ScrollingTreeScrollingNode::commitStateAfterChildren(const ScrollingStateNo
     if (scrollingStateNode.hasChangedProperty(ScrollingStateNode::Property::RequestedScrollPosition))
         handleScrollPositionRequest(scrollingStateNode.requestedScrollData());
 
+    if (scrollingStateNode.hasChangedProperty(ScrollingStateNode::Property::KeyboardScrollData))
```
If you encoded the "stop" state in KeyboardScrollData, you'd wouldn't need the `KeyboardScrollDataEnd` bit.

smfr commented on Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.h:
```diff
@@ -28,6 +28,7 @@
 #if ENABLE(ASYNC_SCROLLING)
 
 #include "IntRect.h"
+#include "KeyboardScroll.h"
```
Prefer forward declaration.

smfr commented on Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.h:
```diff
@@ -59,6 +60,10 @@ class WEBCORE_EXPORT ScrollingTreeScrollingNode : public ScrollingTreeNode {
     virtual ~ScrollingTreeScrollingNode();
 
+    void handleKeyboardScrollRequest(const KeyboardScroll&);
+
```
No blank line.

smfr commented on Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.h:
```diff
@@ -47,6 +47,10 @@ class ScrollingTreeScrollingNodeDelegate {
     virtual void updateFromStateNode(const ScrollingStateScrollingNode&) { }
 
+    virtual void handleKeyboardScrollRequest(const KeyboardScroll&) { }
+
```
No blank line.

smfr commented on Source/WebCore/page/scrolling/mac/ScrollingTreeScrollingNodeDelegateMac.h:
```diff
@@ -51,6 +51,10 @@ class ScrollingTreeScrollingNodeDelegateMac : public ThreadedScrollingTreeScroll
     bool handleWheelEvent(const PlatformWheelEvent&);
 
+    void handleKeyboardScrollRequest(const KeyboardScroll&) override;
+
```
No blank line.

smfr commented on Source/WebCore/platform/KeyboardScrollingAnimator.cpp:
```diff
@@ -284,7 +212,7 @@ bool KeyboardScrollingAnimator::beginKeyboardScrollGesture(ScrollDirection direc
         return false;
 
-    m_currentKeyboardScroll = scroll;
+    m_currentKeyboardScroll = scroll; 
```
Trailing whitespace.

smfr commented on Source/WebCore/platform/KeyboardScrollingAnimator.h:
```diff
@@ -52,21 +52,18 @@ class KeyboardScrollingAnimator : public CanMakeWeakPtr<KeyboardScrollingAnimato
     std::optional<KeyboardScroll> makeKeyboardScroll(ScrollDirection, ScrollGranularity) const;
     float scrollDistance(ScrollDirection, ScrollGranularity) const;
 
     ScrollAnimator& m_scrollAnimator;
```
Can we just reference a `ScrollableArea` here now instead of `ScrollAnimator`?

rr-codes replied:
I don't think so, since `KeyboardScrollingAnimator::beginKeyboardScrollGesture` uses it in several places

smfr replied:
It would be better for KeyboardScrollingAnimator to reference the ScrollableArea, and get the ScrollAnimator from it when it needs one.

smfr commented on Source/WebCore/platform/ScrollAnimationKeyboard.cpp:
```diff
@@ -0,0 +1,249 @@
+        m_velocity = { };
+    }
+
+    return true;
```
This return value is unused.

rr-codes replied:
Yes, this is because `animateScroll` overrides a virtual function, so it is forced to return a value, and in the case of `ScrollAnimationKeyboard`, the value doesn't need to be used

smfr commented on Source/WebCore/platform/ScrollAnimationKeyboard.h:
```diff
@@ -0,0 +1,68 @@
+
+    RectEdges<bool> scrollableDirectionsFromPosition(FloatPoint position);
+
+    std::optional<WebCore::KeyboardScroll> m_currentKeyboardScroll;
```
No need for `WebCore::`

smfr commented on Source/WebCore/platform/ScrollAnimationKeyboard.h:
```diff
@@ -0,0 +1,68 @@
+    FloatSize m_velocity;
+    MonotonicTime m_timeAtLastFrame;
+    FloatPoint m_idealPosition;
+    bool m_scrollTriggeringKeyIsPressed;
```
Put the bool last to optimize padding. Initialize it.

smfr commented on Source/WebCore/platform/ScrollableArea.h:
```diff
@@ -68,6 +69,10 @@ class ScrollableArea : public CanMakeWeakPtr<ScrollableArea> {
     virtual bool isListBox() const { return false; }
 
+    WEBCORE_EXPORT void beginKeyboardScroll(const KeyboardScroll& scrollData);
+
```
No blank line.

smfr commented on Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
```diff
@@ -102,6 +102,10 @@ class AsyncScrollingCoordinator : public ScrollingCoordinator {
     WEBCORE_EXPORT void frameViewWillBeDetached(FrameView&) override;
 
+    WEBCORE_EXPORT bool requestStartKeyboardScrollAnimation(ScrollableArea&, const KeyboardScroll&) final;
+
```
Remove this blank line.

smfr commented on Source/WebCore/page/scrolling/ScrollingCoordinatorTypes.h:
```diff
@@ -132,6 +133,11 @@ struct RequestedScrollData {
 };
 
+struct RequestedKeyboardScrollData {
+    bool stopImmediately { false };
```
Using a bool makes this hard to read at the call site. Better to add an enum. I'd also put it second (since cancel is the less common action).

smfr replied:
A good change would be to have an `action` enum:
```
enum class KeyboardScrollAction : uint8_t {
  StartAnimation,
  StopWithAnimation,
  StopImmediately
}
```

smfr commented on Source/WebCore/page/scrolling/ScrollingStateScrollingNode.cpp:
```diff
@@ -199,6 +200,13 @@ void ScrollingStateScrollingNode::setSynchronousScrollingReasons(OptionSet<Synch
 #endif
 
+
+void ScrollingStateScrollingNode::setKeyboardScrollData(const RequestedKeyboardScrollData& scrollData)
```
This could be use `RequestedKeyboardScrollData&&` with `WTFMove` at the call site and `WTFMove` into the member variable.

rr-codes committed:
Scrolling by pressing space bar has regressed in trunk, is choppy (not 120Hz)
https://bugs.webkit.org/show_bug.cgi?id=246878
rdar://101052130

Reviewed by Simon Fraser.

Moves keyboard smooth scrolling onto the scrolling thread instead of the main thread.

A new `ScrollAnimation` is created to facilitate this, `ScrollAnimationKeyboard`. Most
of the changes are to allow the state to propogate through the various classes and
the scrolling tree.

Since keyboard scrolling is now on a different thread, some tests had to be modified to
enable the `AsyncOverflowScrollingEnabled` and `AsyncFrameScrollingEnabled` settings.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::requestStartKeyboardAnimation):
(WebCore::FrameView::requestFinishKeyboardAnimation):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp:
(WebCore::AsyncScrollingCoordinator::requestStartKeyboardAnimation):
(WebCore::AsyncScrollingCoordinator::requestFinishKeyboardAnimation):
* Source/WebCore/page/scrolling/AsyncScrollingCoordinator.h:
* Source/WebCore/page/scrolling/ScrollingCoordinator.h:
(WebCore::ScrollingCoordinator::requestStartKeyboardAnimation):
(WebCore::ScrollingCoordinator::requestFinishKeyboardAnimation):
* Source/WebCore/page/scrolling/ScrollingStateNode.h:
* Source/WebCore/page/scrolling/ScrollingStateScrollingNode.cpp:
(WebCore::ScrollingStateScrollingNode::ScrollingStateScrollingNode):
(WebCore::ScrollingStateScrollingNode::setKeyboardScrollData):
(WebCore::ScrollingStateScrollingNode::setKeyboardScrollEndData):
* Source/WebCore/page/scrolling/ScrollingStateScrollingNode.h:
(WebCore::ScrollingStateScrollingNode::keyboardScrollData const):
(WebCore::ScrollingStateScrollingNode::keyboardScrollEndData const):
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp:
(WebCore::ScrollingTreeScrollingNode::commitStateAfterChildren):
(WebCore::ScrollingTreeScrollingNode::handleKeyboardScrollRequest):
(WebCore::ScrollingTreeScrollingNode::handleKeyboardScrollEndRequest):
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.h:
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.h:
(WebCore::ScrollingTreeScrollingNodeDelegate::handleKeyboardScrollRequest):
(WebCore::ScrollingTreeScrollingNodeDelegate::handleKeyboardScrollEndRequest):
* Source/WebCore/page/scrolling/mac/ScrollingTreeScrollingNodeDelegateMac.h:
* Source/WebCore/page/scrolling/mac/ScrollingTreeScrollingNodeDelegateMac.mm:
(WebCore::ScrollingTreeScrollingNodeDelegateMac::handleKeyboardScrollRequest):
(WebCore::ScrollingTreeScrollingNodeDelegateMac::handleKeyboardScrollEndRequest):
* Source/WebCore/platform/KeyboardScrollingAnimator.cpp:
(WebCore::KeyboardScrollingAnimator::KeyboardScrollingAnimator):
(WebCore::KeyboardScrollingAnimator::beginKeyboardScrollGesture):
(WebCore::KeyboardScrollingAnimator::handleKeyUpEvent):
(WebCore::KeyboardScrollingAnimator::stopScrollingImmediately):
(WebCore::perpendicularAbsoluteUnitVector): Deleted.
(WebCore::KeyboardScrollingAnimator::updateKeyboardScrollPosition): Deleted.
(WebCore::farthestPointInDirection): Deleted.
(WebCore::KeyboardScrollingAnimator::stopKeyboardScrollAnimation): Deleted.
* Source/WebCore/platform/KeyboardScrollingAnimator.h:
* Source/WebCore/platform/ScrollAnimation.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/ScrollAnimation.h:
* Source/WebCore/platform/ScrollAnimationKeyboard.cpp: Added.
(WebCore::perpendicularAbsoluteUnitVector):
(WebCore::boxSideForDirection):
(WebCore::farthestPointInDirection):
(WebCore::ScrollAnimationKeyboard::scrollableDirectionsFromPosition):
(WebCore::ScrollAnimationKeyboard::ScrollAnimationKeyboard):
(WebCore::ScrollAnimationKeyboard::retargetActiveAnimation):
(WebCore::ScrollAnimationKeyboard::serviceAnimation):
(WebCore::ScrollAnimationKeyboard::startKeyboardScroll):
(WebCore::ScrollAnimationKeyboard::stopKeyboardScrollAnimation):
(WebCore::ScrollAnimationKeyboard::finishKeyboardScroll):
(WebCore::ScrollAnimationKeyboard::animateScroll):
(WebCore::ScrollAnimationKeyboard::debugDescription const):
* Source/WebCore/platform/ScrollAnimationKeyboard.h: Copied from Source/WebCore/platform/ScrollAnimation.cpp.
* Source/WebCore/platform/ScrollAnimator.cpp:
(WebCore::ScrollAnimator::ScrollAnimator):
* Source/WebCore/platform/ScrollableArea.cpp:
(WebCore::ScrollableArea::beginKeyboardScroll):
(WebCore::ScrollableArea::endKeyboardScroll):
* Source/WebCore/platform/ScrollableArea.h:
(WebCore::ScrollableArea::requestStartKeyboardAnimation):
(WebCore::ScrollableArea::requestFinishKeyboardAnimation):
* Source/WebCore/platform/ScrollingEffectsController.cpp:
(WebCore::ScrollingEffectsController::animationCallback):
(WebCore::ScrollingEffectsController::startKeyboardScroll):
(WebCore::ScrollingEffectsController::finishKeyboardScroll):
(WebCore::ScrollingEffectsController::updateKeyboardScrollingAnimatingState):
(WebCore::ScrollingEffectsController::scrollAnimationWillStart):
* Source/WebCore/platform/ScrollingEffectsController.h:
(WebCore::ScrollingEffectsControllerClient::updateKeyboardScrollPosition): Deleted.
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::requestStartKeyboardAnimation):
(WebCore::RenderLayerScrollableArea::requestFinishKeyboardAnimation):
* Source/WebCore/rendering/RenderLayerScrollableArea.h:

Canonical link: https://commits.webkit.org/256266@main
